SU2_CFD/src/solvers/CSpeciesSolver.cpp

```diff
@@ -0,0 +1,529 @@
+                                         unsigned long ExtIter) {
+  const bool Restart = (config->GetRestart() || config->GetRestart_Flow());
+
+  /// NOTE TK:: Check whether this is done in the constructor for example and whether this is mnecessary. The TurbSolver
+  /// does not do this!
+  /*--- For a restart do nothing here. Otherwise initialize the solution with the Init value. ---*/
+  if ((!Restart) && ExtIter == 0) {
+    if (rank == MASTER_NODE) {
+      cout << "Initializing passive scalar (initial condition)." << endl;
+      cout << "initialization = " << nVar << " " << config->GetSpecies_Init(0) << endl;
+    }
+
+    // TK:: arbitrary number which would be easy to recognize during debugging
+    su2double scalar_init[MAXNVAR] = {1234.0};
+
+    for (unsigned long iMesh = 0; iMesh <= config->GetnMGLevels(); iMesh++) {
+      for (unsigned long iPoint = 0; iPoint < geometry[iMesh]->GetnPoint(); iPoint++) {
+        for (unsigned long iVar = 0; iVar < nVar; iVar++) Solution[iVar] = 0.0;
+
+        for (int iVar = 0; iVar < nVar; iVar++) {
+          scalar_init[iVar] = config->GetSpecies_Init(iVar);
+        }
+
+        solver_container[iMesh][SPECIES_SOL]->GetNodes()->SetSolution(iPoint, scalar_init);
+      }
+
+      solver_container[iMesh][SPECIES_SOL]->InitiateComms(geometry[iMesh], config, SOLUTION);
+      solver_container[iMesh][SPECIES_SOL]->CompleteComms(geometry[iMesh], config, SOLUTION);
+
+      solver_container[iMesh][FLOW_SOL]->InitiateComms(geometry[iMesh], config, SOLUTION);
+      solver_container[iMesh][FLOW_SOL]->CompleteComms(geometry[iMesh], config, SOLUTION);
+
+      solver_container[iMesh][FLOW_SOL]->Preprocessing(geometry[iMesh], solver_container[iMesh], config, iMesh,
+                                                       NO_RK_ITER, RUNTIME_FLOW_SYS, false);
+    }
+  }
+}
+
+void CSpeciesSolver::Viscous_Residual(unsigned long iEdge, CGeometry* geometry, CSolver** solver_container,
+                                      CNumerics* numerics, CConfig* config) {
+  /*--- Define an object to set solver specific numerics contribution. ---*/
+  auto SolverSpecificNumerics = [&](unsigned long iPoint, unsigned long jPoint) {
+    /*--- Mass diffusivity coefficients. ---*/
+
+    numerics->SetDiffusionCoeff(nodes->GetDiffusivity(iPoint), nodes->GetDiffusivity(jPoint));
+  };
+
+  /*--- Now instantiate the generic implementation with the functor above. ---*/
+
+  Viscous_Residual_impl(SolverSpecificNumerics, iEdge, geometry, solver_container, numerics, config);
+}
+
+void CSpeciesSolver::BC_Inlet(CGeometry* geometry, CSolver** solver_container, CNumerics* conv_numerics,
+                              CNumerics* visc_numerics, CConfig* config, unsigned short val_marker) {
+  /// NOTE TK:: This is a strong impl whereas TurbSA and inceuler implement a weak version. Testing required.
+  su2double* inlet_species = new su2double[nVar]{};
```

Comment:
static, also currently leaking.

Reply:
this inlet stuff is now changed altogether ...